### ISSUE TITLE
Issue when popper overflows viewport

### ISSUE THREAD

**Issue by chrisdrackett** [souporserious/react-popper#16]
I'm not sure if this is a popper issue or a react-popper issue, so I'll report here now as I keep digging.

I have a popper with a fixed size (width and height). When I open it (placement=bottom) in a window where there is room, I don't have any issues, its placed correctly.

https://cl.ly/1w0m2T1P1j3u

If I think resize the window to the point that the popper no longer fits, things get odd. Basically from what I can tell the `position: absolute` is removed from the `Popper` styles. All the other values are still there, so I'm not sure why or what is going on.

https://cl.ly/1t1M44063V2J

If I then resize the window so there is again size, it still no longer has `position: absolute`:

https://cl.ly/1W3A1c3T1x3C

here is what I'm seeing in the DOM:

```
<div style="top: 0px; left: 0px; transform: translate3d(-380px, 64px, 0px); will-change: transform;" data-placement="bottom"><div data-css-32nwhk="View.Column.Content"><div data-css-1hqne8e="View.Column">Content!</div></div></div>
```

**Comment by chrisdrackett** [comment]
FYI this seems to be related to `preventOverflow`. I need to get more familiar with `Popper.js` itself before I completely understand what is going on, but I am still surprised that when I give the room back, the component does not snap back into its "correct" position.

**Comment by souporserious** [comment]
@FezVrasta any idea on what's going on here?

**Comment by souporserious** [comment]
This just started happening to me 😕 `position: absolute` gets removed randomly, not sure what is causing it.

**Comment by FezVrasta** [comment]
Do you define the position manually or take what Popper provides?

**Comment by souporserious** [comment]
Yeah, it happens if I use `placement="bottom"`. Right when it starts to go out of the viewport, `position: absolute` gets removed.

**Comment by FezVrasta** [comment]
I wasn't clear, I wondered if the position was defined by react-popper or by Popper.js directly.
https://github.com/souporserious/react-popper/blob/master/src/Popper.jsx#L146

Seems like it's on Popper.js side. May you open a ticket there pls?

**Comment by FezVrasta** [comment]
It should be fixed in 1.9.1

**Comment by larrydahooster** [comment]
I am facing the same issue with missing 'position: absolute' after a flip event.

I dig deeper into the code and found the problem that causes the missing 'position' attribute. I am not sure if this is the right repository to address the issue cause it references to popper.js itself https://github.com/FezVrasta/popper.js/blob/v1.9.1/src/popper/modifiers/flip.js#L119 

![screen shot 2017-05-10 at 18 12 05](https://cloud.githubusercontent.com/assets/4989851/25909074/4dc9019a-35ac-11e7-89f6-5130d51d6241.png)


In that line it tries to read a position attribute from data.instance.state but as you can see there is no position on that state, and I don't see any other reference in the popper code that adresses, references or sets state.position. Can you guys help me? @FezVrasta @souporserious. Am I setting it up wrong? Is there a problem between react-popper and popper.js or is it a bug? I really would love to see this issue solved cause I really like the popper.js library.

Thanks in advance.

**Comment by FezVrasta** [comment]
The bug was fixed on Popper.js side a while ago
https://github.com/FezVrasta/popper.js/commit/5e8529cc5d796d6607c86f3b7689e448fb3e1bed (1.9.1)

**Comment by larrydahooster** [comment]
Thx for your quick reply, I still do see a bug in the flip modifier cause it tries to access `data.instance.state.postion` on the data parameter that you just manipulated before `data.offsets.popper.position = 'absolute'; `

So wouldn't it be correct to also read the position from data.offsets.popper.position?

**Comment by FezVrasta** [comment]
Popper.js now doesn't use anything different from `absolute` anymore, the code could use some cleanup.

**Comment by larrydahooster** [comment]
Okay, thx for your input. I just monkeyPatched it for me to work. Maybe I find some time to contribute a cleanup on popper.js

**Comment by FezVrasta** [comment]
I'm on it.

**Comment by souporserious** [comment]
Sorry about that! I've been super busy. I'll try and give this some TLC today and get the outstanding issues on my side resolved.

**Comment by FezVrasta** [comment]
@larrydahooster this https://github.com/FezVrasta/popper.js/commit/ae06fabbc1d2f9e386f54a6333a3d2d1c3fe8cd4 should avoid any unwanted side effect. Waiting for the CI to pass and I'll publish a patch version.

**Comment by FezVrasta** [comment]
1.9.2 is out, please let me know if it helps

**Comment by larrydahooster** [comment]
Thx for your quick fix. I can't test it now since I left the office. Gonna check it tomorrow morning. Tanks again, I really appreciate it. Nice Framework.

**Comment by souporserious** [comment]
@FezVrasta is it possible you can pull master and run the demo? It looks like this issue is happening all of the time now after upgrading Popper.js 😞

**Comment by FezVrasta** [comment]
I can take a look tomorrow, I'm sorry :(

**Comment by souporserious** [comment]
Any chance to look at this @FezVrasta? I'd like to hold off on a new release until we can determine what is going on here.

**Comment by souporserious** [comment]
@FezVrasta forcing `position: absolute` seems to fix everything, when would it not be `absolute`?

**Comment by FezVrasta** [comment]
At the moment it will always be `absolute`, but in the future I may want to apply optimizations that will require a different `position`.

I fixed the problem on 1.9.5

**Comment by souporserious** [comment]
Thanks! Seems to be working great. Going to close this, thanks again 🎉
